packages/carbon-web-components/src/components/toggle/toggle.ts

```diff
@@ -150,15 +159,21 @@ class CDSToggle extends HostListenerMixin(CDSCheckbox) {
         value="${ifDefined(value)}"
         ?disabled=${disabled}
         id="${id}"></button>
-      <label for="${id}" class="${prefix}--toggle__label">
-        <span class="${prefix}--toggle__label-text">${labelText}</span>
-        <div class="${inputClasses}">
-          <div class="${toggleClasses}" @click=${handleChange}>
+      <label for="${id}" class="${prefix}--toggle__label" part="label">
+        <span class="${prefix}--toggle__label-text" part="label-text"
+          >${labelText}</span
+        >
+        <div class="${inputClasses}" part="appearance">
+          <div
+            class="${toggleClasses}"
+            part="switch switch--checked"
```

Comment:
We should only add `switch--checked` when `checked` is true.
```suggestion
            part="switch${checked ? ' switch--checked' : ''}"
```